File: data/readable_format/cpp/file1084.cpp
#include <iostream>
#include <omp.h>

void foo(int &r);

int main() {
    int r = 0;
    foo(r);

    if (r == 11) {
        std::cout << "Test passed, r = " << r << std::endl;
    } else {
        std::cerr << "Test failed, r = " << r << std::endl;
        return 1; // Indicate an error
    }

    return 0; // Indicate success
}

void foo(int &r) {
    #pragma omp parallel
    {
        #pragma omp single
        {
            #pragma omp parallel for reduction(+:r)
            for (int i = 1; i <= 10; ++i) {
                r = r + 1;
            }
            r = r + 1;
        }
    }
}

File: data/readable_format/fortran/file1084.f90
program test
  implicit none
  integer :: r

  r = 0
  call foo (r)

  if (r == 11) then
    print *, "Test passed, r = ", r
  else
    print *, "Test failed, r = ", r
    call abort
  end if

contains

  subroutine foo (r)
    integer :: i, r
    !omp parallel
    !omp single
    !omp target teams distribute parallel do reduction (+: r)
      do i = 1, 10
        r = r + 1
      end do
      r = r + 1
    !omp end single
    !omp end parallel
  end subroutine

end program test